### PR TITLE
Update SSL config documentation for SSL profiles and m_ssl_mbedtls.

### PR DESCRIPTION
**Pull request by SaberUK**

### CODE REVIEW

File: docs/conf/inspircd.conf.example [hunk 1]
```diff
@@ -177,11 +177,13 @@
...
-# on each end of the link without any significant issues.
-# Supported SSL types are: "openssl" and "gnutls".
-# You must load the ssl_openssl module for OpenSSL or ssl_gnutls for GnuTLS.
+# When linking servers, the OpenSSL, GnuTLS, and mbedTLS implementations are
+# completely link-compatible and can be used alongside each other on each end
+# of the link without any significant issues.
+# Supported SSL types are:
+#   "openssl" (requires the ssl_openssl module)
+#   "gnutls" (requires the ssl_gnutls module)
+#   "mbedtls" (requires the ssl_mbedtls module)
```
Comment (attilamolnar) [inline docs/conf/inspircd.conf.example]: This is not the case anymore, ssl profiles are given a name in their tag and referenced by that name in `<link>` or `<bind>`, for example:

    <sslprofile name="strong" provider="openssl" ciphers="foo" certfile="cert1.pem" keyfile="...">
    <sslprofile name="normal" provider="openssl" ciphers="bar" certfile="cert2.pem" keyfile="...">
    <sslprofile name="normal-websocket" provider="openssl" ciphers="baz" certfile="cert2.pem" keyfile="..." requestclientcert="false">
    <bind address="..." port="6697" type="clients" ssl="normal">
    <bind address="..." port="7002" type="clients" hook="websocket" ssl="normal-websocket">
    <bind address="..." port="7000,7001" type="servers" ssl="strong">

The `<gnutls>`, `<openssl>` and `<mbedtls>` tags are only used if there is no `<sslprofile>` tag in the config, to retain compatibility with configs from 2.0.

Comment (attilamolnar) [inline docs/conf/inspircd.conf.example]: This is not documented anywhere currently, the documentation of SSL modules in the config files is essentially "check out the wiki for the available config options" in 2.0. Perhaps we should keep that and show a basic configuration with a ssl profiles.

**Comment by SaberUK** [comment]
@attilamolnar Hows this?

File: docs/conf/inspircd.conf.example [hunk 2]
```diff
@@ -153,9 +153,12 @@
       type="clients"
 
       # ssl: If you want the port(s) in this bind tag to use SSL, set this
-      # to either "gnutls" or "openssl". The appropriate SSL module must be
-      # loaded for SSL to work. If you do not want the port(s) in this bind
-      # tag to support SSL, just remove or comment out this option.
+      # to "openssl", gnutls", "mbedtls" or the name of a custom <sslprofile>
+      # tag that you have defined. See the wiki page for the SSL module you are
+      # using for more details.
```
Comment (attilamolnar) [inline docs/conf/inspircd.conf.example]: Not quite, the "openssl" etc. tags are only intended for backwards compat so they only work if there are no sslprofiles defined (https://github.com/inspircd/inspircd/blob/master/src/modules/extra/m_ssl_openssl.cpp#L842)

File: docs/conf/inspircd.conf.example [hunk 3]
```diff
@@ -152,10 +152,13 @@
       # to this bind section.
       type="clients"
 
-      # ssl: If you want the port(s) in this bind tag to use SSL, set this
-      # to either "gnutls" or "openssl". The appropriate SSL module must be
-      # loaded for SSL to work. If you do not want the port(s) in this bind
-      # tag to support SSL, just remove or comment out this option.
+      # ssl: If you want the port(s) in this bind tag to use SSL, set this to
+      # the name of a custom <sslprofile> tag that you have defined or one
+      # of "openssl", gnutls", "mbedtls" if you have not defined any. See the
```
Comment (attilamolnar) [inline docs/conf/inspircd.conf.example]: Opening " is missing for gnutls

File: docs/conf/links.conf.example [hunk 4]
```diff
@@ -36,13 +36,16 @@
       # failover (see above).
       timeout="300"
 
-      # ssl: If defined, this states the SSL module that will be used when
-      # making an outbound connection to the server. Options are: "openssl"
-      # and "gnutls" (they are compatible with each other).
+      # ssl: If defined, this states the SSL profile that will be used when
+      # making an outbound connection to the server. Options are the name of an
+      # <sslprofile> tag that you have defined or one of "openssl", gnutls",
```
Comment (attilamolnar) [inline docs/conf/links.conf.example]: Same here

File: docs/conf/links.conf.example [hunk 5]
```diff
@@ -36,13 +36,16 @@
...
 
-      # ssl: If defined, this states the SSL module that will be used when
-      # making an outbound connection to the server. Options are: "openssl"
-      # and "gnutls" (they are compatible with each other).
+      # ssl: If defined, this states the SSL profile that will be used when
+      # making an outbound connection to the server. Options are the name of an
+      # <sslprofile> tag that you have defined or one of "openssl", "gnutls",
+      # "mbedtls" if you have not defined any. See the wiki page for the SSL
+      # module you are using for
+      # more details.
```
Comment (SaberUK) [inline docs/conf/links.conf.example]: How did I miss that…

Comment (SaberUK) [inline docs/conf/links.conf.example]: Fixed!

**Comment by attilamolnar** [comment]
Thanks!

**Pull request merged**